### ISSUE TITLE
Operations allow me to specify an interface as a return type. Well. Kind of. Should they?

### ISSUE THREAD

**Issue by johanste** [microsoft/cadl#1026]
Only the openapi emitter complains about operations returning interfaces. But otherwise it seem to compile just fine. Is that intentional? 

[Playground Link](https://cadlplayground.z22.web.core.windows.net/?c=aW50ZXJmYWNlIFRoaXNJc05vdEFNb2RlbCB7Cgp9CgpvcCB5b2xvKCk60CE7Cg%3D%3D)

**Comment by timotheeguerin** [comment]
Think this is the same base problem as #561

**Comment by markcowl** [comment]
- fix in the compiler

**Comment by markcowl** [comment]
pri: 1
est: 5

**Comment by tjprescott** [comment]
@markcowl, questions on this. 
1. should the label be Required for DPG 2.0?
2. Tim says it's the same problem as #561 but that is on the backlog. Just confirming this is something we want to do this sprint.
3. Are we saying this is a bug (it's not labeled as such)? When you say "fix in compiler" do you mean that this should emit an error (warning?) when this happens?

**Comment by johanste** [comment]
This is not required for DPG 2.0.

**Comment by tjprescott** [comment]
> This is not required for DPG 2.0.

Then is this really Pri 1 for this sprint?

**Comment by markcowl** [comment]
Yeah, the only reason this was marked as 'required' was in case it heralded a larger problem - if Timothee identified the problem, then this should no longer be required.  I downgraded it in the sprint.

**Comment by burdiyan** [comment]
I think typespec itself should allow this, to be able to model capability-based RPC like Cap’n Proto, and capnweb.dev. Each individual emitter should probably complain when they don’t support this pattern.

**Comment by burdiyan** [comment]
@timotheeguerin I see this has been closed as completed, but there's no reference to any commit or PR. I wonder what the final decision about this was.

**Comment by timotheeguerin** [comment]
Keeping the status quo, this is not up to the TypeSpec language to be opiniated by this. Protocol libraries like http might decide this is not something that is allowed in their scope however.

**Comment by burdiyan** [comment]
Great! Thanks a lot. I was worried about never being able to use TypeSpec to express Cap'n Web / Cap'n Proto interfaces, but I guess that would indeed be possible if someone writes the necessary package.
